x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -191,10 +196,8 @@ public void startDeployment(TrainedModelDeploymentTask task, ActionListener<Trai
                     NlpTask nlpTask = new NlpTask(nlpConfig, vocabulary);
                     NlpTask.Processor processor = nlpTask.createProcessor();
                     processContext.nlpTaskProcessor.set(processor);
-                    // here, we are being called back on the searching thread, which MAY be a network thread
-                    // `startAndLoad` creates named pipes, blocking the calling thread, better to execute that in our utility
-                    // executor.
-                    executorServiceForDeployment.execute(() -> processContext.startAndLoad(modelConfig.getLocation(), modelLoadedListener));
+
+                    executeStartAndLoad.onResponse(modelConfig);
```

Comment:
There's no need to make this change. `executeStartAndLoad` is only called from this one place, and not as a listener for an async action. You can move the code from `executeStartAndLoad` back to here and get rid of that listener.